articles/kotlin-basics/index.md

```diff
@@ -0,0 +1,1701 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/kotlin-basics/
+title: Kotlin Programming for Beginners 
+description: Learn Kotlin Programming, its basics, and Fundamentals from scratch.
+author: stephen-chacha
+date: 2021-01-12T00:00:00-14:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/kotlin-basics/hero.png
+    alt: kotlin basics image
+---
+
+#### Introduction
+Learn Kotlin Programming, its basics, and Fundamentals from scratch. 
+
+#### Prerequisites
+For you to follow along with this tutorial, you need to have:
+- Kotlin editor preferably [IntelliJ](https://www.jetbrains.com/idea/download/#section=windows) installed on your machine.
+
+### Topics to be covered
+1. Data Types and Variables 
+    - Data Types and Variables
+    - String, Literals, and String Interpolation
+    - Comments 
+
+2. Constants, Variables, and Data Types
+
+3. Control Flow Statements 
+    - IF ELSE 
+    - IF Expressions
+    - WHEN Expressions
+
+4. Loop Control Statements 
+    - What are Iterators?
+    - FOR Loop and how it works
+    - WHILE Loop
+    - DO WHILE Loop 
+    - BREAK statements 
+    - CONTINUE keyword 
+    - Labelled FOR Loop 
+
+5. Functions and Interoperability 
+    - Declaring functions 
+    - Interoperability with Java code 
+    - Function as Expressions 
+    - Extension Functions
+    - Infix Functions
+    - Default Parameters
+    - Named Parameters
+    - Tailrec Functions 
+
+6. Object Oriented Programming in Kotlin 
+    - Defining Class and creating Objects
+    - INIT block
+    - Primary and Secondary Constructors 
+    - Properties ( Field variables )
+    - Inheritance
+    - Method and Property Overriding 
+    - Polymorphism 
+    - Abstract Class, Property and Method
+    - Interface 
+    - Data Class
+    - Object Declaration
+    - Companion Object
+
+7. Functional Programming in Koltin
+    - Lambdas
+    - Higher-Order Functions
+    - Closures
+    - 'it' keyword
+    - 'with' function
+    - 'apply' function
+
+8. Collections in Kotlin
+    - Arrays
+    - List
+    - Map and HashMap
+    - Set and HashSet  
+
+9. Sorting and Filtering
+    - "filter" function
+    - "map" function
+    - Predicates: all, any, find, count. 
+
+10. Kotlin NULL Safety
+    - Safe call
+    - with Let
+    - Elvis
+    - Lateinit keyword
+    - Lazy delegation and 'lateinit' vs. 'lazy'
+
+
+```kotlin
+// Hello World 
+fun main(args: Array<String>) {
+    print("Hello World")
+}
+
+// Output: Hello world
+```
+
+### Defining Variables
+
+#### `val`
+It can be assigned a value only once 
```

Comment:
Give a more description of the keyword. Since you said it's learning kotlin from scratch, please provide a more detailed definition of it and what it does.